test/Test/Database/LSMTree/Normal/StateMachine.hs

```diff
@@ -1176,8 +1176,29 @@ shrinkActionWithVars ::
   -> [Any (LockstepAction (ModelState h))]
 shrinkActionWithVars _ _ = \case
     New p conf -> [ Some $ New p conf' | conf' <- QC.shrink conf ]
-    Inserts kins tableVar -> [ Some $ Inserts kins' tableVar | kins' <- QC.shrink kins ]
+
+    -- Shrink inserts and deletes towards updates.
+    Updates upds tableVar -> [
+        Some $ Updates upds' tableVar
+      | upds' <- QC.shrink upds
+      ]
+    Inserts kvbs tableVar -> [
+        Some $ Inserts kvbs' tableVar
+      | kvbs' <- QC.shrink kvbs
+      ] <> [
+        Some $ Updates (V.map f kvbs) tableVar
+      | let f (k, v, mb) = (k, R.Insert v mb)
+      ]
```

Comment:
This will first shrink to the smallest insert/delete and then switch to update. Is that the order you want, rather than switching to update first and then shrink as an update, or if not possible then shrink as insert/delete.